src/remote_callbacks.rs

```diff
@@ -87,6 +89,14 @@ pub type PushTransferProgress<'a> = dyn FnMut(usize, usize, usize) + 'a;
 ///     * total
 pub type PackProgress<'a> = dyn FnMut(PackBuilderStage, usize, usize) + 'a;
 
+/// Callback used to inform of upcoming updates.
+///
+/// The argument is a slice containing the updates which will be sent as
+/// commands to the destination.
+///
+/// The return value indicates whether the push should continue.
+pub type PushNegotiation<'a> = dyn FnMut(&[PushUpdate<'_>]) -> bool + 'a;
```

Comment:
Is there any particular reason this returns a bool? I was thinking it could return a `Result<(), Error>`, similar to other callbacks so that it is clear that it is triggering an error. A bool seems more innocuous, and not a typical way to do error handling in Rust.

Reply:
Thanks for the review! I've made the changes. I might have been emulating `sideband_progress`, where the callback also returns a `bool` to decide whether to cancel the operation, but using a `Result` is indeed more idiomatic here.